moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js

```diff
@@ -0,0 +1,126 @@
+import React from "react";
+import { TextField, CircularProgress } from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import { v4 as uuidv4 } from "uuid";
+
+const signalToFeatureCollection = signal => {
+  // Immitate a "drawn point" feature from a traffic signal feature
+  // If no signal, returns featureCollection with no features (resets fc state)
+  let featureCollection = {
+    type: "FeatureCollection",
+    features: [],
+  };
+
+  if (signal) {
+    const featureUUID = uuidv4();
+    // todo: actually store the traffic signal ID somewhere
+    // probably as new component attributes "asset_id" and "asset_type"
+    const feature = {
+      type: "Feature",
+      properties: {
+        ...signal.properties,
+        renderType: "Point",
+        PROJECT_EXTENT_ID: featureUUID,
+        sourceLayer: "drawnByUser",
+      },
+      geometry: signal.geometry,
+      id: featureUUID,
+    };
+    featureCollection.features.push(feature);
+  }
+  return featureCollection;
+};
+
+const SignalAutocomplete = ({
```

Comment:
If/when we want to re-use this in the components editor—we'll need to wrap some of this project create form-specific handling. tbd?